Reflection/Ground.simba

```diff
@@ -10,21 +10,52 @@ Example:
 *)
 Function R_GetMainscreenGroundItemByID(ItemID: Int32): Array of RSTile;
 var
-  I, W, H: Integer;
+  I: Integer;
   ResultTile: RSTile;
   Items: Array of RSTile;
+  MSBounds: TBox;
 begin
   Items := RSGroundObject.GetTileByID(ItemID);
-  if Length(Items) < 1 then
+  MSBounds := R_MainscreenBounds;
+  if High(Items) < 1 then
     Exit;
 
-  for I := 0 to Length(Items) - 1 do
+  for I := 0 to High(Items) do
   begin
-    GetClientDimensions(W, H);
     ResultTile := RSTile(Items[i]).ToLocal.TileToMS;
-    if (not ((ResultTile.X < 0) or (ResultTile.X > W) or (ResultTile.Y < 0) or (ResultTile.Y > H))) then
+    if (not ((ResultTile.X > MSBounds.X1) and (ResultTile.X < MSBounds.X2) and (ResultTile.Y > MSBounds.Y1) and (ResultTile.Y < MSBounds.Y2))) then
```

Comment:
Something i use in my scripts:

```
function TPoint.InBox(Box: TBox): Boolean; constref;
begin
  Result := PointInBox(Self, Box);
end; 
```
Then you can just ask ResultTile.InBox(MSBounds)

Reply:
I'm surprised that I forgot about that function.. Thanks!